src/main/java/org/automation/listeners/TestRunListener.java

```diff
@@ -66,6 +75,22 @@ public void onStart(ISuite suite) {
     @Override
     public void onFinish(ISuite suite) {
         Log.info("About to end executing Suite [" + suite.getName() + "]");
+
+    }
+
+    private int getPassPercentage(ITestContext context) {
+        int totalTests = context.getAllTestMethods().length;
+        int passedTests = context.getPassedTests().size();
+        return (passedTests * 100) / totalTests;
     }
 
+    @Override
+    public boolean retry(ITestResult result) {
+        if (retryCount < maxRetryCount) {
```

Comment:
Improvement Suggestion: Include a mechanism to handle known intermittent failures based on specific exceptions if possible.